bin/oneoff/backfill_data/school_info_id

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env ruby
+
+# This script copies user_id-school_info_id pairs from the pd_enrollments
+# table into the school_info_id column in users.
+
+require_relative '../../../dashboard/config/environment'
+
+# build hash mapping school_info_id to school_id
+school_info_id_school_id_mapping = {}
+
+puts "PROCESSING SCHOOL_INFO"
+SchoolInfo.find_each(batch_size: 2500) do |school_info|
+  school_info_id_school_id_mapping[school_info.id] = school_info.school_id
+  print "."
+end
+puts "\nPROCESSED SCHOOL_INFO"
+
+batch_number = 0
+User.find_in_batches(batch_size: 2_500) do |batch|
+  puts "PROCESSING USERS: #{batch_number}..."
+  ActiveRecord::Base.transaction do
+    batch.each do |user|
+      # only do this if user has no school_info_id or their school_info_id is
+      # not associated with a school_id
+      # (don't do it if the user already has a school_info_id associated with a school_id)
+      next if school_info_id_school_id_mapping[user.school_info_id]
+
+      enrollments = Pd::Enrollment.for_user(user)
+      next if enrollments.empty?
+
+      enrollments_with_school_id = enrollments.select {|e| school_info_id_school_id_mapping[e.school_info_id]}
+
+      # if the user has pd_enrollments whose school_info_id is associated with
+      # a school_id, update the user with the most recent of those pd_enrollments' school_info_id
+      if !enrollments_with_school_id.empty?
+        enrollment = enrollments_with_school_id.sort_by {|e| e[:updated_at]}.last
+        user.update(school_info_id: enrollment.school_info_id)
```

Comment:
@aoby Joanne pointed out that in a lot of scripts we save users with `user.save(validate: false)` in case there is something wrong with the user. Do you have an opinion on this? Should I make that change to this script?

Reply:
(Sigh) Yes, that appears to be the case. (e.g. https://github.com/code-dot-org/code-dot-org/blob/8d961a3524e323e867e2d42e31a16ced3a8f4691/lib/cdo/school_info_interstitial_helper.rb#L22-L25). Probably best to follow the pattern, though we really should clean up our users data :(

Also we should be using the bang methods (`user.update!(...)`) so we'll see an error if it fails. The non-bang versions will return a bool. This is less important once we're skipping validations here but still worth knowing if sql is unreachable or something else unexpected.